server/routes/volunteers.ts

```diff
@@ -144,6 +144,441 @@ router.route("/api/volunteers/").get(
+      });
+
+      return;
+    }
+
+    const volunteerRecordId = volunteerRecordIdFetch.records[0].id;
+
+    // to update persons info issue patch to the Volunteers CRM table
+    const updatePersonInfoBody = {
+      records: [
+        {
+          id: volunteerRecordId,
+          fields: {
+            "First Name": firstName,
+            "Last Name": lastName,
+            "Email Address": email,
+            "Phone Number": phoneNumber,
+          },
+        },
+      ],
+    };
+
+    const contactInfoUpdateResult = await airtablePATCH({
+      url: `${AIRTABLE_URL_BASE}/🙋🏽Volunteers CRM`,
+      body: updatePersonInfoBody,
+    });
+
+    if (contactInfoUpdateResult.kind === "error") {
+      res.status(INTERNAL_SERVER_ERROR).json({
+        message: contactInfoUpdateResult.error,
+      });
+
+      return;
+    }
+
+    if (
+      !hasVolunteerTypeChanged(
+        airtableParticipantType,
+        originalInfo.records[0].fields.Type
+      )
+    ) {
+      logger.info(
+        `Volunteer ${volunteerId} has not changed their volunteer type.`
+      );
+      res.status(OK).json({
+        message: `Volunteer ${volunteerId} has not changed their volunteer type.`,
+      });
+
+      return;
+    }
+
+    const supplierPickupEvent =
+      originalInfo.records[0].fields["🚛 Supplier Pickup Event"];
+
+    const originalRolesSet = new Set([...originalInfo.records[0].fields.Type]);
+    const newRolesSet = new Set(airtableParticipantType);
+
+    const wasBoth =
+      originalRolesSet.has("Driver") && originalRolesSet.has("Distributor");
+    const isNowBoth =
+      newRolesSet.has("Driver") && newRolesSet.has("Distributor");
+    const isNewlySingleRole =
+      newRolesSet.size === 1 &&
+      (newRolesSet.has("Driver") || newRolesSet.has("Distributor"));
+
+    let slotPayload;
+
+    // cases
+    // 1. Volunteer had only one role and is now choosing both roles.
+    // 2. Volunteer had only one role and is switching to the other single role.
+    // 3. Volunteer had both roles and is still keeping both roles.
+    // 4. Volunteer had one role and is keeping that same role.
+    // 5. Volunteer had both roles and is now choosing only one role.
+
+    // need to do a time check here also
+    const driverLookupSlot =
+      `${AIRTABLE_URL_BASE}/⏳ Available Slots?` +
+      `filterByFormula=AND(` +
+      `{Time Slot (readable)}="10:30am",` +
+      `{🚛 Supplier Pickup Event Record ID}="${supplierPickupEvent}",` +
+      `Type="Driver"
+      )`;
+
+    const logisticsLookupSlot =
+      `${AIRTABLE_URL_BASE}/⏳ Available Slots?` +
+      `filterByFormula=AND(` +
+      `{🚛 Supplier Pickup Event Record ID}="${supplierPickupEvent}",` +
+      `Type="Distributor"
+      )`;
+
+    if (wasBoth) {
+      if (isNewlySingleRole) {
+        // Scenario 5: They were originally both and now only one.
+        let roleToNullify = newRolesSet.has("Driver")
+          ? "Driver"
+          : "Distributor";
```

Comment:
Should this be flipped?

Reply:
You are correct it should be flipped, the reason this worked was later down when determining the slots the ternary flipped it back around. I clarified this in the new commit